nixos/doc/manual/development/systemd-environment-and-secrets.section.md

```diff
@@ -0,0 +1,32 @@
+## Environment and secret maangement {#sec-systemd-environment}
+
+The `environment` attribute is used to pass environment variables to the systemd service.
+```
+environment = {
+  MY_ENV_VAR = "Something blue";
+}
+```
+
+A file can be passed as well with `environmentFile`. If both are passed, `environmentFile` takes precedence over `environment`
+
+Environment variables should not contain secret data.
+The main reason being that environment variables are accessible to unpriviledged users.
+See [systemd doc](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Environment=) for a more detailed explanation.
+
+### Secret management {#sec-systemd-secrets}
+
+To pass secret data to a systemd service, systemd has the `LoadCredential` mechanism.
+By passing a file path to systemd, it will make sure that the file is available to the process under a specific path.
+The load credential attribute takes an id a colon and a file path
+let's have a look at an [example](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-apps/lemmy.nix#L182)
```

Comment:
Maybe a permalink like this `https://github.com/NixOS/nixpkgs/blob/3b0bff383f8e894b1ee7a30b62c8b727a68511df/nixos/modules/services/web-apps/lemmy.nix#L182` is better?

Reply:
Thinking about those, I thought it would be better to just include the code-snippet rather than a link.
My next iteration will just have the relevant snippets just copied.